writer/ynab/ynab.go

```diff
@@ -36,7 +37,37 @@ type Ytransactions struct {
 	Transactions []Ytransaction `json:"transactions"`
 }
 
-func ynabberToYNAB(cfg ynabber.Config, t ynabber.Transaction) Ytransaction {
+// importIDMaker tries to return a unique YNAB import ID to avoid duplicate
+// transactions.
+func importIDMaker(cfg ynabber.Config, t ynabber.Transaction) string {
+	// Version 1 uses the memo, amount and date from Ytransaction
+	v1Cutover := time.Time(cfg.YNAB.ImportID.V1)
+	v1 := func(t ynabber.Transaction) string {
+		date := t.Date.Format("2006-01-02")
+		amount := t.Amount.String()
+		hash := sha256.Sum256([]byte(t.Memo))
+		return fmt.Sprintf("YBBR:%s:%s:%x", amount, date, hash[:2])
+	}
+
+	// Version 2 uses the account and transaction ID from ynabber.Transaction
+	v2Cutover := time.Time(cfg.YNAB.ImportID.V2)
+	v2 := func(t ynabber.Transaction) string {
+		hash := sha256.Sum256([]byte(t.Account.ID + t.ID))
```

Comment:
Please add date and amount as part of the hash as well. They do not hurt, are certain to remain the same, and if, for whatever reason they were to ever change, this should result in a new transaction where the user will resolve this in YNAB. I don't think this kind of situation will ever be possible but we cannot know how different banks operate. Furthermore, we do not know how the transaction id is generated for different banks. Is it even unique ...

A specific example: with credit card transactions, there might be a much higher amount 'reserve' first which is later corrected with the right amount. I think in my case, the reservation is never imported, but we can't guarantee the operation is similar in all banks.

( To be honest, the particular case for reservation, if we see these, they should not be imported at all. But as it is, we do not know how to filter these out so this would have to be done if we get to know about these kinds of transactions appearing. )

Reply:
Done! 

Regarding the reserved transactions, I believe that they should be part of the "Pending" slice: https://github.com/frieser/nordigen-go-lib/blob/main/accounts.go#L76 and thus not be imported until they are "Booked": https://github.com/martinohansen/ynabber/blob/22906008c8606c77dfc94743a8344e92e8866c3f/reader/nordigen/nordigen.go#L109

Reply:
Possible. In reality, we would really need extensive test data from multiple banks in different situations.

Perhaps we could, at some point, include a tool which will allow users to use ynabber to fetch Nordigen data, anonymize it (i.e. mangle all identification and private data) and sent it as test data.